### ISSUE TITLE
Should deriving `InvariantFree` require that a structs fields are the same pub level as the struct?

### ISSUE THREAD

**Issue by rylev** [rust-secure-code/mem-markers#18]
Since `InvariantFree` types are meant to represent types without invariants, it should reason that all the fields of the type should be accessible. Should we require that a structs fields are the same visibility level as the struct itself?

**Comment by RustyYato** [comment]
No, it should be fine for the fields to be more private than the struct. There doesn't seem to be any reason to make this restriction.
